Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsFragmentPresenter.java

```diff
@@ -31,11 +31,16 @@
 
 import java.io.File;
 import java.util.ArrayList;
+import java.util.LinkedHashMap;
+import java.util.Map;
 
 public final class SettingsFragmentPresenter
 {
   private SettingsFragmentView mView;
 
+  public static final LinkedHashMap<String, String> LOG_TYPE_NAMES =
+          NativeLibrary.GetLogTypeNames();
```

Comment:
Is this guaranteed to run after `NativeLibrary.Initialize`? I don't believe it will work correctly otherwise.

Reply:
From my testing I haven't seen any problems here. Should I add a method that sets this LinkedHashMap and call it with `AfterDirectoryInitializationRunner.run` immediately after `NativeLibrary.Initialize` for an abundance of caution?

Reply:
Please see the latest commit. Let's (hopefully) prevent a lot of problems with calling `NativeLibrary` by completing initialization before launching Settings and Emulation activities.

Reply:
I'm still not sure if there is any guarantee that things will execute in the right order. The information I can find is that `static` variables will be initialized when the class is loaded, but I can't find anything about when specifically classes actually are loaded.

If you are going to gate launching the settings activity behind directory initialization, please keep in mind that there already is a check for that here, and I would prefer only having one check for it for the settings (but perhaps your new location for it is better?): https://github.com/dolphin-emu/dolphin/blob/63c53ebc87bbdb3c917858669becb4d5a1148110/Source/Android/app/src/main/java/org/dolphinemu/dolphinemu/features/settings/ui/SettingsActivityPresenter.java#L85

Reply:
I put a breakpoint on `NativeLibrary.GetLogTypeNames` and indeed it initialized when the class was loaded. The class is only (eventually) called through `SettingsActivity.launch`. I may be missing something but it seems this shouldn't cause any problems. I'll look into the second part of your response soon.

Reply:
Reverted the last commit. We eventually call `DirectoryInitialization.areDolphinDirectoriesReady()` for both Settings and Emulation fragments. Also, `SettingsActivity.launch` adds a static context so it isn't worth changing the check to there.